How to perform tensor parallelism when `vocab_size` is not an integer multiple of `world_size`?

When I try to load a bloom-based model to an 8-GPUs server, I met the following error:

```
AssertionError: 250682 is not divisible by 8
```

It seems the following check will be performed before tensor parallelism: https://github.com/vllm-project/vllm/blob/c894836108732d0cbb6fce15aeda8de1218a380d/vllm/model_executor/parallel_utils/tensor_parallel/utils.py#L66-L70

I check the model's `config.json` and found its `vocab_size` is 250682. Obviously it cannot pass the check above.

So how could I perform tensor parallelism when `vocab_size` is not an integer multiple of `world_size`?

--- a/vllm/model_executor/models/bloom.py
+++ b/vllm/model_executor/models/bloom.py
@@ -285,7 +285,7 @@ class BloomForCausalLM(nn.Module):
         for name, loaded_weight in hf_model_weights_iterator(
                 model_name_or_path, cache_dir, use_np_cache):
             if name == "lm_head.weight":
-                # Since hidden_states are parallelized, we need to 
+                # Since hidden_states are parallelized, we need to
                 # load lm_head.weight in parallel.
                 self._column_parallel_weights.append(name)
                 # If lm_head is provided, use it instead.

--- a/vllm/model_executor/models/bloom.py
+++ b/vllm/model_executor/models/bloom.py
@@ -286,13 +286,15 @@ class BloomForCausalLM(nn.Module):
                 model_name_or_path, cache_dir, use_np_cache):
             # If lm_head is provided in weights, use it instead.
             if name == "lm_head.weight":
-                self.lm_head_weight.data.copy_(state_dict["lm_head_weight"])
-                continue
+                # Since hidden_states are parallelized, we need to 
+                # load lm_head.weight in parallel.
+                self._column_parallel_weights.append(name)
+                param = state_dict["lm_head_weight"]
+            else:
+                if not name.startswith("transformer."):
+                    name = "transformer." + name
+                param = state_dict[name]
 
-            if not name.startswith("transformer."):
-                name = "transformer." + name
-
-            param = state_dict[name]
             if "query_key_value" in name:
                 # NOTE(woosuk): BLOOM's fused QKV has the shape of
                 # [num_heads * 3 * head_size, hidden_size], while the

--- a/vllm/model_executor/models/bloom.py
+++ b/vllm/model_executor/models/bloom.py
@@ -25,6 +25,7 @@ from typing import Dict, List, Optional, Tuple
 
 import torch
 from torch import nn
+from torch.nn.parameter import Parameter
 from transformers import BloomConfig
 
 from vllm.model_executor.input_metadata import InputMetadata
@@ -284,11 +285,13 @@ class BloomForCausalLM(nn.Module):
         state_dict = self.state_dict()
         for name, loaded_weight in hf_model_weights_iterator(
                 model_name_or_path, cache_dir, use_np_cache):
+            # If lm_head is provided in weights, use it instead.
             if name == "lm_head.weight":
-                name = "lm_head_weight"
-            else:
-                if not name.startswith("transformer."):
-                    name = "transformer." + name
+                self.lm_head_weight = Parameter(state_dict["lm_head_weight"])
+                continue
+
+            if not name.startswith("transformer."):
+                name = "transformer." + name
 
             param = state_dict[name]
             if "query_key_value" in name:

--- a/vllm/model_executor/models/bloom.py
+++ b/vllm/model_executor/models/bloom.py
@@ -284,8 +284,11 @@ class BloomForCausalLM(nn.Module):
         state_dict = self.state_dict()
         for name, loaded_weight in hf_model_weights_iterator(
                 model_name_or_path, cache_dir, use_np_cache):
-            if not name.startswith("transformer."):
-                name = "transformer." + name
+            if name == "lm_head.weight":
+                name = "lm_head_weight"
+            else:
+                if not name.startswith("transformer."):
+                    name = "transformer." + name
 
             param = state_dict[name]
             if "query_key_value" in name:

--- a/vllm/model_executor/models/bloom.py
+++ b/vllm/model_executor/models/bloom.py
@@ -25,7 +25,6 @@ from typing import Dict, List, Optional, Tuple
 
 import torch
 from torch import nn
-from torch.nn.parameter import Parameter
 from transformers import BloomConfig
 
 from vllm.model_executor.input_metadata import InputMetadata
@@ -287,7 +286,7 @@ class BloomForCausalLM(nn.Module):
                 model_name_or_path, cache_dir, use_np_cache):
             # If lm_head is provided in weights, use it instead.
             if name == "lm_head.weight":
-                self.lm_head_weight = Parameter(state_dict["lm_head_weight"])
+                self.lm_head_weight.data.copy_(state_dict["lm_head_weight"])
                 continue
 
             if not name.startswith("transformer."):

--- a/vllm/model_executor/models/bloom.py
+++ b/vllm/model_executor/models/bloom.py
@@ -284,12 +284,12 @@ class BloomForCausalLM(nn.Module):
         state_dict = self.state_dict()
         for name, loaded_weight in hf_model_weights_iterator(
                 model_name_or_path, cache_dir, use_np_cache):
-            # If lm_head is provided in weights, use it instead.
             if name == "lm_head.weight":
                 # Since hidden_states are parallelized, we need to 
                 # load lm_head.weight in parallel.
                 self._column_parallel_weights.append(name)
-                param = state_dict["lm_head_weight"]
+                # If lm_head is provided, use it instead.
+                param = self.lm_head_weight
             else:
                 if not name.startswith("transformer."):
                     name = "transformer." + name

